client/apis/volumesnapshot/v1/types.go

```diff
@@ -0,0 +1,429 @@
+
+// VolumeSnapshotClass specifies parameters that a underlying storage system uses when
+// creating a volume snapshot. A specific VolumeSnapshotClass is used by specifying its
+// name in a VolumeSnapshot object.
+// VolumeSnapshotClasses are non-namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:printcolumn:name="Driver",type=string,JSONPath=`.driver`
+// +kubebuilder:printcolumn:name="DeletionPolicy",type=string,JSONPath=`.deletionPolicy`,description="Determines whether a VolumeSnapshotContent created through the VolumeSnapshotClass should be deleted when its bound VolumeSnapshot is deleted."
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+type VolumeSnapshotClass struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// driver is the name of the storage driver that handles this VolumeSnapshotClass.
+	// Required.
+	Driver string `json:"driver" protobuf:"bytes,2,opt,name=driver"`
+
+	// parameters is a key-value map with storage driver specific parameters for creating snapshots.
+	// These values are opaque to Kubernetes.
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters"`
+
+	// deletionPolicy determines whether a VolumeSnapshotContent created through
+	// the VolumeSnapshotClass should be deleted when its bound VolumeSnapshot is deleted.
+	// Supported values are "Retain" and "Delete".
+	// "Retain" means that the VolumeSnapshotContent and its physical snapshot on underlying storage system are kept.
+	// "Delete" means that the VolumeSnapshotContent and its physical snapshot on underlying storage system are deleted.
+	// Required.
+	DeletionPolicy DeletionPolicy `json:"deletionPolicy" protobuf:"bytes,4,opt,name=deletionPolicy"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeSnapshotClassList is a collection of VolumeSnapshotClasses.
+// +kubebuilder:object:root=true
+type VolumeSnapshotClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// items is the list of VolumeSnapshotClasses
+	Items []VolumeSnapshotClass `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeSnapshotContent represents the actual "on-disk" snapshot object in the
+// underlying storage system
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:subresource:status
```

Comment:
Currently, the storage and served version `true` is set to the v1beta1. v1 should have `served: true` and `storage: true` and v1beta1 should only have `served: true`. To do that
```suggestion
// +kubebuilder:subresource:status
// +kubebuilder:storageversion
```